src/Authentication/AuthenticationManager.php

```diff
@@ -103,7 +104,7 @@ public function getAccount(RequestInterface $request, $cache = TRUE) {
 
     if (!$account) {
 
-      if ($this->plugins->count() && !$this->getIsOptional()) {
+      if (RestfulManager::isRestfulPath($request) && $this->plugins->count() && !$this->getIsOptional()) {
```

Comment:
> What was the use case?

I wanted the user to be switched to the one authenticated by RESTful upon ``hook_file_download_access()`` -- so on private files an anonymous user can send ``https://example.com/system/files/some-private-file.jpg?access_token=12345``

```php
/**
 * Implements hook_file_download_access().
 *
 * Get the user account from the RESTful authentication.
 */
function shoov_restful_file_download_access($file_item, $entity_type, $entity) {
  $auth_plugins = array_keys(restful_get_authentication_plugins());

  $auth_manager = new RestfulAuthenticationManager();

  // Register all authentication plugins.
  foreach ($auth_plugins as $auth_plugin_name) {
    $auth_handler = restful_get_authentication_handler($auth_plugin_name);
    $auth_manager->addAuthenticationProvider($auth_handler);
    $auth_manager->setIsOptional(TRUE);
  }

  $request = restful_parse_request();
  $auth_manager->getAccount($request);
}
```